addons/crm_survey/models/survey_user_input.py

```diff
@@ -0,0 +1,134 @@
+                    row_value = answer_id.matrix_row_id.display_name
+                    col_value = answer_id.suggested_answer_id.display_name
+                    description += '<br/>&emsp;' + f"{row_value} — {col_value}"
+
+                # Others
+                else:
+                    if first_answer:
+                        description += " — "
+                    answer = answer_id.display_name
+                    if first_answer:
+                        description += str(answer)
+                        first_answer = False
+                    else:
+                        description += ', ' + str(answer)
+
+                # Check if answer should create a lead
+                if not is_lead_answer and answer_id.suggested_answer_id and answer_id.suggested_answer_id.create_lead:
+                    is_lead_answer = True
+
+                # Check if the question has a nickname recorded
+                if answer_id.question_id.save_as_nickname:
+                    user_nickname = answer
+
+                # Check if the question has a email answer
+                if answer_id.question_id.validation_email:
+                    public_user_mail = answer
+            description += "</li></ul>"
+
+            ### Generate the lead
+            if is_lead_answer:
+                self._generate_lead(user_input, user_nickname, description, public_user_mail)
+
+    def _generate_lead(self, user_input, user_nickname, description, public_user_mail):
+        """ This method will:
+        - generate an new opportunity
+        - link that to the current survey
+        """
+        medium = self.env['utm.medium']._fetch_or_create_utm_medium('Survey')
+
+        source = self.env['utm.source'].search([('name', '=', self.survey_id.title)])
+        if not source:
+            source = self.env['utm.source'].create({
+                        'name': self.survey_id.title,
+                    })
+
+        # Check if the survey responsible is from a sales team
+        survey_responsible = user_input.survey_id.user_id
+        if survey_responsible:
+            is_survey_responsible_in_sales_team = self.env['crm.team'].search([('member_ids', 'in', survey_responsible.id)])
+            if is_survey_responsible_in_sales_team:
+                survey_responsible = survey_responsible.id
+            else:
+                survey_responsible = False
+        else:
+            survey_responsible = False
+
+        # Get the username
+        username = user_input.partner_id.name
+        if not username:  # Public user
+            username = "Participant#" + str(user_input.id)
+            if user_nickname:
+                lead_title_start = f"{user_nickname}'s"
+            elif public_user_mail:
+                lead_title_start = f"{public_user_mail}'s"
+            else:
+                lead_title_start = "New"
+        else:
+            lead_title_start = f"{username}'s"
+        contact_name = user_nickname or username
```

Comment:
you can inline this in the dict below :) no need for the variable I would say ;)